ratis-netty/src/main/java/org/apache/ratis/netty/server/DataStreamManagement.java

```diff
@@ -231,23 +232,48 @@ StreamInfo remove(Key key) {
     return f;
   }
 
+
+  private IdentityHashMap<RaftPeer, RaftPeer> getRouteTable(RaftClientRequest request)
```

Comment:
Let's create a org.apache.ratis.protocol.RouteTable interface/class in ratis-common.
```
 interface RoutingTable {
    List<RaftPeerId> getSuccessors(RaftPeerId peer);
  }
```
I think we need RaftPeerId since the RaftPeer can be looked up locally.

BTW, why using IdentityHashMap but not just a Map in the implementation?

Reply:
IdentityHashMap allows duplicated key, such as primary -> peer1, primary -> peer2, we can put them in IdentityHashMap, but can not put them in a map which do not allow duplicated key.

Reply:
IdentityHashMap may have unexpected behaviour since it uses == but not equals(..).  Let's use 
```
Map<RaftPeerId, List<RaftPeerId>>
```
Or, we may use guava Multimap https://guava.dev/releases/19.0/api/docs/com/google/common/collect/Multimap.html

Reply:
Got it, I will use Map<RaftPeerId, List<RaftPeerId>>. Thanks a lot.